contracts/cw3-multiple-choice/src/msg.rs

```diff
@@ -0,0 +1,165 @@
+    pub refund_failed_proposals: Option<bool>,
+    /// The existing governance token address
+    pub gov_token_address: String,
+    /// The parent dao contract address
+    pub parent_dao_contract_address: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    /// Makes a new proposal
+    Propose(ProposeMsg),
+    /// Vote on an open proposal
+    Vote(VoteMsg),
+    /// Execute a passed proposal
+    Execute { proposal_id: u64 },
+    /// Close a failed proposal
+    Close { proposal_id: u64 },
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    /// Return ThresholdResponse
+    Threshold {},
+    /// Returns ProposalResponse
+    Proposal {
+        proposal_id: u64,
+    },
+    /// Returns ProposalListResponse
+    ListProposals {
+        start_after: Option<u64>,
+        limit: Option<u32>,
+    },
+    /// Returns ProposalListResponseb
+    ReverseProposals {
+        start_before: Option<u64>,
+        limit: Option<u32>,
+    },
+    /// Returns the number of proposals in the DAO (u64)
+    ProposalCount {},
+    // Returns config
+    GetConfig {},
+    /// Returns VoteResponse
+    Vote {
+        proposal_id: u64,
+        voter: String,
+    },
+    /// Returns VoteListResponse
+    ListVotes {
+        proposal_id: u64,
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// Returns information about current tallys for a
+    /// proposal. Returns type `VoteTallyResponse`.
+    Tally {
+        proposal_id: u64,
+    },
+    /// Returns VoterInfo
+    Voter {
+        address: String,
+    },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct VoteMsg {
+    pub proposal_id: u64,
+    pub vote: Vote,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct ProposeMsg {
+    pub title: String,
+    pub description: String,
+    pub choices: Vec<String>,
+    pub msgs: Vec<Vec<CosmosMsg<Empty>>>,
+    // note: we ignore API-spec'd earliest if passed, always opens immediately
+    pub latest: Option<Expiration>,
+}
+
+/// This defines the different ways tallies can happen.
+///
+/// The total_weight used for calculating success as well as the weights of each
+/// individual voter used in tallying should be snapshotted at the beginning of
+/// the block at which the proposal starts (this is likely the responsibility of a
+/// correct cw4 implementation).
+/// See also `ThresholdResponse` in the cw3 spec.
+/// This is similar to thresholds for 'yes' votes in single choice voting, but in this case in order for any
+/// of the multiple choice options to pass, it must have threshold votes in its favor.
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum Threshold {
```

Comment:
There are a ton of subtle details we've found in the new implementation that you'll likely want to move in here. See this comment for an overview: https://github.com/DA0-DA0/dao-contracts/blob/00e9b4d70848c4f79c519dd9a9f184373f99681c/contracts/cw-govmod-single/src/threshold.rs#L7-L38

I can make another pass later and go over the logic here for that sort of stuff.